core/templates/dev/head/player/PlayerServices.js

```diff
@@ -315,7 +315,9 @@ oppia.factory('oppiaPlayerService', [
                 newStateName, LearnerParamsService.getAllParams());
             }
           }
-
+          if (_editorPreviewMode) {
```

Comment:
ok. I'm fine with b), so I've removed this now, but I definitely think we should keep track of the state in preview mode so it can be returned to in the gui view.  Would I need to call editorContextService.setActiveStateName() in the same place I was changing the path, and also for the case where the back/forward buttons are clicked?